docs/user/private-app-repository.md

```diff
@@ -202,6 +202,18 @@ The above will generate a Pod with the label `my-repo: isPrivate` and the enviro
 
 ## Per Namespace App Repositories
 
-Previously, once an App Repository was created in Kubeapps, the charts of that repository were then available cluster-wide to all users of Kubeapps. This was changed to allow creating App Repositories available only in specific namespaces, enabling future work supporting deploying charts with private docker registries. You can still create an App Repository whose charts will be available to users in all namespaces by selecting "All Namespaces" when creating the repository.
+Previously, once an App Repository was created in Kubeapps, the chart of that repository were then available cluster-wide to all users of Kubeapps. This was changed to allow creating App Repositories available only in specific namespaces, enabling future work supporting deploying charts with private docker registries. These can be created by anyone with the required RBAC for that namespace. You can still create an App Repository whose charts will be available to users in all namespaces by selecting "All Namespaces" when creating the repository.
+
+You can give specific users the ability to create App Repositories in a specific namespace by granting them the necessary RBAC:
+
+```bash
+kubectl -n custom-namespace create rolebinding username-apprepositories-read --user username --clusterrole kubeapps:apprepositories-write
+```
+
+or other users the ability to deploy charts from App Repositories in a specific namespace by granting them read access:
+
+```bash
+kubectl -n custom-namespace create rolebinding username-apprepositories-read --user username --clusterrole kubeapps:apprepositories-read
+```
```

Comment:
we already have those roles per namespace:

https://github.com/kubeapps/kubeapps/blob/master/chart/kubeapps/templates/apprepository-rbac.yaml#L68

It is also documented in docs/user/access-control.md . We should use the same roles in this guide.

Reply:
> we already have those roles per namespace:
> 
> https://github.com/kubeapps/kubeapps/blob/master/chart/kubeapps/templates/apprepository-rbac.yaml#L68

Yes, I was going to delete those, as unless I'm missing something, they are no longer useful. They are created in the release namespace only as they are intended to be bound in the release namespace only (ie. before per-namespace app repositories, we provided these unbound roles so people could easily add app repo readers / writers *in the release namespace only*.) Now we similarly want to provide unbound roles which can be applied in any namespace, hence the new cluster roles.

> 
> It is also documented in docs/user/access-control.md . We should use the same roles in this guide.

Depending whether you agree with the above, I'll update that doc to use the new cluster roles as well.

Reply:
ah, right, I missed the cluster/namespace context. To avoid a breaking change we can have both, the role and the clusterrole and modify the documentation to point to that clusterrole.

One thing we need to handle is several instances of Kubeapps per cluster. The easiest thing to do is to include the namespace in the name (even if it's a clusterrole): `kubeapps:{ns}:apprepositories-write`. WDYT?

Reply:
> ah, right, I missed the cluster/namespace context. To avoid a breaking change we can have both, the role and the clusterrole and modify the documentation to point to that clusterrole.

Done. I initially was updating the `access-control.md` to per-namespace instructions, but then realised we shouldn't do that until we switch the default. For now I've just switched the reference to the cluster role.

> 
> One thing we need to handle is several instances of Kubeapps per cluster. The easiest thing to do is to include the namespace in the name (even if it's a clusterrole): `kubeapps:{ns}:apprepositories-write`. WDYT?

I'd originally removed the release name, because it seemed redundant: helm doesn't resource-count anyway, so deleting a chart just deletes all its resources which I thought meant it would delete the (cluster-wide) CRD too, breaking other releases, but I checked and found that deleting a helm release (with helm 3) does *not* delete any CRD's it creates. So yep, I agree - I'll update them to be unique to the release.

Reply:
Note that this also required removing the condition (the feature flag), so that now this RBAC is always added if `.Values.rbac.create` is true.